maven-scm-api/src/main/java/org/apache/maven/scm/ChangeSet.java

```diff
@@ -552,6 +609,16 @@ public String toXML()
         buffer.append( "\t\t<msg><![CDATA[" )
             .append( removeCDataEnd( comment ) )
             .append( "]]></msg>\n" );
+        List<String> tags = getTags();
+        if ( tags != null && !tags.isEmpty() )
```

Comment:
The null check is redudant, see your new code.

Reply:
Fixed